### ISSUE TITLE
Performance in SerializeListener

### ISSUE THREAD

**Issue by bjoernHeneka** [api-platform/core#1966]
Hey there,
didn't found any matching topic.
Maybe someone can help here. I found out that the SerializeListener takes a lot of time.
It just returns a 2 items within a list Resource.

Did i miss sth.?

![api_platform](https://user-images.githubusercontent.com/10229990/40419919-00401030-5e87-11e8-9864-09204df62034.png)

**Comment by IckleChris** [comment]
From what I understand this is only really an issue during development, where the serialization configuration is parsed on each request (to ensure changes made are reflected without having to clear the cache). In production mode the config is cached, and greatly reduces the serialization time. On top of that, more performance improvements are coming in Symfony 4.1 - https://symfony.com/blog/new-in-symfony-4-1-faster-serializer

**Comment by bjoernHeneka** [comment]
Thx. for the fast response @IckleChris. Even in prod env it's faster but still about 200-300ms.

**Comment by dunglas** [comment]
We still have to add support for the new interface you mentioned in API platform. I'm working on it.

**Comment by bjoernHeneka** [comment]
I'm new to blackfire... Hopefuly i did it right here 😄 

https://blackfire.io/profiles/f47693ff-a197-49ad-9646-48cf88140aec/graph

**Comment by dunglas** [comment]
@bjoernHeneka it looks like your cache is not warmed.

1. Be sure to run `composer install --no-dev --classmap-authoritative` before running Blackfire
2. Run the request one time to warm the cache, then run Blackfire

**Comment by bjoernHeneka** [comment]
Hey @dunglas the hint with the classmap did help.. a lot 😇 
But for some reason is not every request fast... about every 4th or 5th is 400ms the others take <20ms...

Nevertheless... thx for the support

**Comment by alanpoulain** [comment]
Performance improvements have been done in the last version of API Platform.
Please open a new issue if the problem is still here.

**Comment by FireLizard** [comment]
This is still an open issue.
I've updated Symfony to the latest LTS, which is v4.4 and api-platform to 2.5

**Performance** (Prod is close to Dev)
![Bildschirmfoto am 2020-10-01 um 18 46 48](https://user-images.githubusercontent.com/7585164/94839214-f9b34880-0416-11eb-8a37-fb2ab2efbd84.png)

Are there any news on this?

**Comment by dunglas** [comment]
@FireLizard could you do a Blackfire profile? It would help a lot to see which method consumes what.

**Comment by FireLizard** [comment]
Hi @dunglas ,
sorry for the late responds.

Here is a Blackfire profile of my local host:
https://blackfire.io/profiles/480b6e9e-3f13-453f-ac9f-d02af6e2725a/graph

**Comment by FireLizard** [comment]
Resolved the bottleneck. Thank you all ;)

**Comment by garciasdos** [comment]
@FireLizard what was your bottleneck reason? I'm having same problems in both dev and prod environments and I do not find the solution 😿

**Comment by FireLizard** [comment]
Hi @garciasdos ,
I've analyzed the application with Blackfire. It helped me a lot to figure out what was the bottleneck during serialization. 
It turns out that on creation of thumbnails the check if its missing from the filesystem has taken way too long. So I created a type of cache-warmup for thumbnail creation.
The thing is: I didn't see that before because the creation took 0.5s (fictional). On production with much more images and an empty cache it quickly reached ~30s. This little differences of execution time between parts of your application are shown really good in Blackfire
